### ISSUE TITLE
tests.regressions.threads.threads_all_1422 failed at Perlmutter 

### ISSUE THREAD

**Issue by weilewei** [STEllAR-GROUP/hpx#6194]
tests.regressions.threads.threads_all_1422 failed for both gcc@11.2.0 and clang@16.0.0 due to the same error. This is the only 1 tests failed out of 974. 

test 'num_cores == os_threads' failed in function 'int hpx_main()': '128' != '10'

## Expected Behavior

tests.regressions.threads.threads_all_1422 passed.

## Actual Behavior

wwei@nid001008:~/src/hpx2/build_clang_Release> cat /global/homes/w/wwei/src/hpx2/build_clang_Release/Testing/Temporary/LastTest.log
Start testing: Mar 10 16:07 PST
----------------------------------------------------------
951/974 Testing: tests.regressions.threads.threads_all_1422
951/974 Test: tests.regressions.threads.threads_all_1422
Command: "/usr/bin/python" "/global/homes/w/wwei/src/hpx2/build_clang_Release/bin/hpxrun.py" "/global/homes/w/wwei/src/hpx2/build_clang_Release/bin/threads_all_1422_test" "-e" "0" "-t" "-1" "-v" "--"
Directory: /global/homes/w/wwei/src/hpx2/build_clang_Release/tests/regressions/threads
"tests.regressions.threads.threads_all_1422" start time: Mar 10 16:07 PST
Output:
----------------------------------------------------------
Cores: 128
OS Threads: 10
/global/homes/w/wwei/src/hpx2/tests/regressions/threads/threads_all_1422.cpp(29): test 'num_cores == os_threads' failed in function 'int hpx_main()': '128' != '10'
0 sanity checks and 1 test failed.
Base command is "/global/homes/w/wwei/src/hpx2/build_clang_Release/bin/threads_all_1422_test --hpx:threads=all"
Executing command: /global/homes/w/wwei/src/hpx2/build_clang_Release/bin/threads_all_1422_test --hpx:threads=all
<end of output>
Test time =   0.09 sec
----------------------------------------------------------
Test Failed.
"tests.regressions.threads.threads_all_1422" end time: Mar 10 16:07 PST
"tests.regressions.threads.threads_all_1422" time elapsed: 00:00:00
----------------------------------------------------------

End testing: Mar 10 16:07 PST

## Steps to Reproduce the Problem

... Please be as specific as possible while describing how to reproduce your problem.

  1.
  1.
  1.

## Specifications

... Please describe your environment

  - HPX Version: latest master, 15a04a7d3a2079592586e6619964b162ddf4e927
  - Platform (compiler, OS):AMD EPYC 7763, both both gcc@11.2.0 and clang@16.0.0

**Comment by hkaiser** [comment]
@weilewei what happens if you launch this test manually while specifying the full number of cores:
```
/global/homes/w/wwei/src/hpx2/build_clang_Release/bin/threads_all_1422_test --hpx:threads=128
```
?

**Comment by weilewei** [comment]
@hkaiser thanks for sharing!

Interestingly, if I a run the test on login node, I don't need to supply any value ` --hpx:threads=XX` to make the test passed. `~/src/hpx2/build_clang_Release> ./bin/threads_all_1422_test ` just passed, but shows 
```
Cores: 256
OS Threads: 256
```
On compute node, I have to supply ` --hpx:threads=128` to make it passed; without it, it only picks up 10 OS threads. 

It seems login node has less CPUs than the compute node. I compiled the code on login node and run the ctest on compute node. Maybe that's the difference?

**Comment by hkaiser** [comment]
> I compiled the code on login node and run the ctest on compute node. Maybe that's the difference?

Yep, you should avoid doing that. The login node might even have a different architecture than the compute nodes, which may lead to segmentation faults and other issues.

**Comment by weilewei** [comment]
The same error is observed after I compiled hpx on compute node.

After compiled hpx on compute node, I run code on compute node immediately:
```
wwei@nid001141:~/src/hpx2/build_gcc_Release> ./bin/threads_all_1422_test 
Cores: 128
OS Threads: 10
/global/homes/w/wwei/src/hpx2/tests/regressions/threads/threads_all_1422.cpp(29): test 'num_cores == os_threads' failed in function 'int hpx_main()': '128' != '10'
0 sanity checks and 1 test failed.
```

and then I ran the same code on login node:
```
wwei@login03:~/src/hpx2/build_gcc_Release> ./bin/threads_all_1422_test 
Cores: 256
OS Threads: 256
```

**Comment by weilewei** [comment]
Thanks for pointing out! I used `salloc ... --cpus-per-task=10 ... " and then I removed `--cpus-per-task=10` to hop on to a compute node. Then I run the test and it passed! 

Thanks so much for the clarification and helping me to find the root cause. Things are clear on my end and feel free to close this ticket if needed.
